components/model_analyzer/TorchBenchAnalyzer.py

```diff
@@ -0,0 +1,120 @@
+
+from components.model_analyzer.dcgm.dcgm_monitor import DCGMMonitor
+from components.model_analyzer.tb_dcgm_types.da_exceptions import TorchBenchAnalyzerException
+from components.model_analyzer.tb_dcgm_types.gpu_device_factory import GPUDeviceFactory
+from components.model_analyzer.dcgm import dcgm_fields
+from components.model_analyzer.tb_dcgm_types.gpu_tensoractive import GPUTensorActive
+from components.model_analyzer.tb_dcgm_types.gpu_utilization import GPUUtilization
+from components.model_analyzer.tb_dcgm_types.gpu_power_usage import GPUPowerUsage
+from components.model_analyzer.tb_dcgm_types.gpu_free_memory import GPUFreeMemory
+from components.model_analyzer.tb_dcgm_types.gpu_used_memory import GPUUsedMemory
+from components.model_analyzer.tb_dcgm_types.gpu_fp32active import GPUFP32Active
+from components.model_analyzer.tb_dcgm_types.record_aggregator import RecordAggregator
+from components.model_analyzer.tb_dcgm_types.tb_logger import set_logger, LOGGER_NAME
+from components.model_analyzer.tb_dcgm_types.config import *
+from components.model_analyzer.tb_dcgm_types.config import DEFAULT_MONITORING_INTERVAL
+
+import logging
+logger = logging.getLogger(LOGGER_NAME)
+
+class ModelAnalyzer:
+    def __init__(self):
+        self.gpu_factory = GPUDeviceFactory()
+        self.gpus = self.gpu_factory.verify_requested_gpus(['all', ])
+        # the metrics to be collected
+        # self.gpu_metrics = [GPUUtilization, GPUPowerUsage,
+        #                     GPUFreeMemory, GPUUsedMemory, GPUFP32Active, GPUTensorActive]
+        self.gpu_metrics = [GPUFP32Active]
+        # the final metric results. Its format is {GPU_UUID: {GPUUtilization: }}
```

Comment:
What is `GPUUtilization` here? Can you give a complete example of the final metric result?

Reply:
`GPUUtilization` is a class imported `from components.model_analyzer.tb_dcgm_types.gpu_utilization`. One example of this `self.gpu_metrics` is 
```
{'GPU-4177e846-1274-84e3-dcde': 
    {<class 'components.model_analyzer.tb_dcgm_types.gpu_fp32active.GPUFP32Active'>: 
            <components.model_analyzer.tb_dcgm_types.gpu_fp32active.GPUFP32Active object at 0x7f14bbae2280>
     }
}
```
The first key is a string indicating the GPU_UUID. The second key is the class of GPUFP32Active. The inside value is an instance of class GPUFP32Active.